File: 2014/S2.py
"""
author: sjay05
"""
from sys import stdin
input = stdin.readline

N = int(input())
l1 = input().split()
l2 = input().split()
prt = {}

if __name__ == '__main__':
    s = True
    for i in range(len(l1)):
        p1 = l1[i]
        p2 = l2[i]
        if p1 in prt.keys():
            if prt[p1] != p2:
                s = False
                break
        if p2 in prt.keys():
            if prt[p2] != p1:
                s = False
                break
        if p1 == p2:
            s = False
            break
        if p1 not in prt.keys(): prt[p1] = p2
        if p2 not in prt.keys(): prt[p2] = p1

    if not s: print "bad"
    else: print "good"

File: 2014/S2.cpp
/*
 * Author: DynamicSquid
 */
 
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main()
{
    int N;
    cin >> N;

    vector<string> students1(N);
    vector<string> students2(N);

    for (int a = 0; a < N; ++a)
    {
        string name;
        cin >> name;
        students1[a] = name;
    }

    for (int a = 0; a < N; ++a)
    {
        string name;
        cin >> name;
        students2[a] = name;
    }

    for (size_t a = 0; a < students1.size(); ++a)
    {
        if (students1[a] == students2[a])
        {
            cout << "bad";
            return 0;
        }

        string partner = students2[a];

        for (size_t b = 0; b < students1.size(); ++b)
        {
            if (students1[b] == partner && students2[b] != students1[a])
            {
                cout << "bad";
                return 0;
            }
        }
    }

    cout << "good";
}
